cookbook/security/securing_application.rst

```diff
@@ -0,0 +1,125 @@
+.. index::
+   single: Security; Best Practices
+
+How to Secure a Symfony Application
+===================================
+
+This article summarizes the steps you should follow to secure your Symfony
+applications.
+
+Configuration
+-------------
+
+Configure the ``secret`` Option
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The value of this option is used to increase the entropy in several security
+related operations. Make sure to change its default value (usually defined in
+the ``app/config/parameters.yml`` file). It's even recommended to change its
+value from time to time.
+
+Read the :ref:`framework.secret documentation <configuration-framework-secret>`
+
+Configure a Strong Password Encryption
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Bcrypt`_ is the most recommended password hash algorithm for modern web
+applications. That's why Symfony only recommends to :ref:`use the bcrypt encoder <reference-security-bcrypt>`
+even for in-memory users defined in the ``app/config/security.yml`` file.
+
+Make your Application HTTPS-only
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If your application defines some protected areas under the ``access_control``
+configuration option in ``app/config/security.yml``, use the ``requires_channel``
+option as explained in :doc:`this article </cookbook/security/force_https>`.
+
+If you prefer to protect some specific routes, define the ``schemes`` route option
+for all those routes as explained in :doc:`this article </cookbook/routing/scheme>`.
+
+Enable or Disabled the CSRF Protection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, Symfony forms include a protection against `CSRF attacks`_. In case
+you need to :ref:`disable this protection for some form <forms-csrf>`, use the
+``csrf_protection`` option. You can even :ref:`disable CSRF protection <reference-csrf_protection-enabled>`
+application-wide.
+
+Furthermore, the form used to log in users doesn't include any CSRF protection.
+Read :doc:`this article </cookbook/security/csrf_in_login_form>` to learn how to
```

Comment:
perhaps this should be expanded to more than login forms since ajax calls are also to be protected by this?